pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+func (rc *AWSresourcesContainer) getSubnetsConfig(
+	res *vpcmodel.MultipleVPCConfigs,
+	subnetNameToNetIntf map[string][]*commonvpc.NetworkInterface,
+	skipByVPC map[string]bool,
+) (vpcInternalAddressRange map[string]*ipblock.IPBlock, err error) {
+	vpcInternalAddressRange = map[string]*ipblock.IPBlock{}
+	for vpcUID := range res.Configs() {
+		vpcInternalAddressRange[vpcUID] = nil
+	}
+	for _, subnet := range rc.SubnetsList {
+		if skipByVPC[*subnet.VpcId] {
+			continue
+		}
+		_, err := commonvpc.UpdateConfigWithSubnet(*subnet.SubnetId,
+			*subnet.SubnetId, *subnet.AvailabilityZone, *subnet.CidrBlock,
+			*subnet.VpcId, res, vpcInternalAddressRange, subnetNameToNetIntf)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return vpcInternalAddressRange, nil
+}
+
+func parseSGTargets(sgResources map[string]map[string]*commonvpc.SecurityGroup,
+	netIntfToSGs map[string][]types.GroupIdentifier,
+	configs *vpcmodel.MultipleVPCConfigs) {
+	for vpcUID, sgs := range sgResources {
+		config := configs.Config(vpcUID)
+		for _, node := range config.Nodes {
+			if node.Kind() == commonvpc.ResourceTypeNetworkInterface {
+				if intfNodeObj, ok := node.(*commonvpc.NetworkInterface); ok {
+					securityGroupIds := netIntfToSGs[intfNodeObj.ResourceUID]
+					for _, securityGroupID := range securityGroupIds {
+						sgs[*securityGroupID.GroupId].Members[intfNodeObj.Address()] = intfNodeObj
+					}
+				}
+			}
+		}
+	}
+}
+
+func (rc *AWSresourcesContainer) getSGconfig(
+	res *vpcmodel.MultipleVPCConfigs,
+	skipByVPC map[string]bool,
+	netIntfToSGs map[string][]types.GroupIdentifier,
+) error {
+	sgMap := map[string]map[string]*commonvpc.SecurityGroup{} // map from vpc uid to map from sg name to its sg object
+	sgLists := map[string][]*commonvpc.SecurityGroup{}
+	for i := range rc.SecurityGroupsList {
+		sg := rc.SecurityGroupsList[i]
+		if skipByVPC[*sg.VpcId] {
+			continue
+		}
+		vpcUID := *sg.VpcId
+		vpc, err := commonvpc.GetVPCObjectByUID(res, vpcUID)
+		if err != nil {
+			return err
+		}
+		commonvpc.NewSGResource(*sg.GroupId, *sg.GroupId, vpc, NewSpecificAnalyzer(sg), sgMap, sgLists)
+	}
+	parseSGTargets(sgMap, netIntfToSGs, res)
+	for vpcUID, sgListInstance := range sgLists {
+		vpc, err := commonvpc.GetVPCObjectByUID(res, vpcUID)
+		if err != nil {
+			return err
+		}
+		sgLayer := &commonvpc.SecurityGroupLayer{
+			VPCResource: vpcmodel.VPCResource{
+				ResourceType: vpcmodel.SecurityGroupLayer,
+				VPCRef:       vpc,
+				Region:       vpc.RegionName(),
+			},
+			SgList: sgListInstance}
+		res.Config(vpcUID).FilterResources = append(res.Config(vpcUID).FilterResources, sgLayer)
+	}
+
+	for _, vpcSgMap := range sgMap {
+		for _, sg := range vpcSgMap {
+			// the name of SG is unique across all SG of the VPC
+			err := sg.Analyzer.PrepareAnalyzer(vpcSgMap, sg)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+/********** Functions used in Debug mode ***************/
+
+func printVPCConfigs(c *vpcmodel.MultipleVPCConfigs) {
```

Comment:
why can't all the printing functions be in `commonvpc`?

Reply:
because it calls printConfigs func which is specific for each provider(for now)

Reply:
can't this function be common for all providers? it iterates over objects from vpcmodel pkg

Reply:
can't, because for ex in ibmvpc this function calls `printTGWAvailableRoutes` which is specific for ibm and includes struct type that is in ibmvpc pkg.. if the struct were defined in commonvpc (it is not used in aws..) it will be possible to move the func to common-code
I can split the func to two functions.. but I think it is better leave it this way..

Reply:
ok, let's consider this for a followup PR